app/src/main/java/com/amaze/filemanager/adapters/data/LayoutElementParcelable.java

```diff
@@ -52,25 +55,25 @@
     //same as hfile.modes but different than openmode in Main.java
     private OpenMode mode = OpenMode.FILE;
 
-    public LayoutElementParcelable(boolean isBack, String goback, boolean showThumbs) {
-        this(true, new File("..").getName(), "..", "", "", goback, 0,
+    public LayoutElementParcelable(@NonNull Context c, boolean isBack, String goback, boolean showThumbs) {
```

Comment:
Context is not written to parcel and yet you expect it in constructor. This might have problems when rotating screen.